app/src/main/java/com/example/cartoon/Notification.kt

```diff
@@ -21,31 +26,36 @@ class Notification(private val context: Context) {
     }
 
     fun sendNotification(title: String, message: String, smallIcon: Int, notificationId: Int) {
-        val notification = NotificationCompat.Builder(context, CHANNEL_ID)
-            .setContentTitle(title)
-            .setContentText(message)
-            .setSmallIcon(smallIcon)
-            .setPriority(NotificationCompat.PRIORITY_DEFAULT)
-            .build()
-
-        val notificationManager = NotificationManagerCompat.from(context)
-        if (ActivityCompat.checkSelfPermission(
-                context,
-                Manifest.permission.POST_NOTIFICATIONS
-            ) != PackageManager.PERMISSION_GRANTED
+        if (ContextCompat.checkSelfPermission(context, Manifest.permission.POST_NOTIFICATIONS)
```

Comment:
I can see a user flow missing here. Let say for the first time if the permission was not present, and android ask for it.
1. you are not handling callback received  received from ActivityCompat.requestPermission() through onActivityResult,  So by default the user won't see a toast message or notification for the first time when the whole process happens.

Currently we are seeing a mesage from saveImage() method  in the activity. 
so the sendNotification have to be called from saveImage().


I would suggest moving the whole permission to activity . 
Keep things here simple 
 if permission exist -> show notification -> else show a toast.

Reply:
will look into this

Reply:
so do you want it like this. as soon as the app opens after splash screen the premission dialog should appear??

Reply:
I have one one more doubt like if the user allows notification permission the n also toast should be shown?? Like image saved.

Reply:
yes the toast must be shown. I guess notification will make more sense if it has some functionality like when clicked it takes the user to where the image is saved in gallery because right now it does nothing.

Reply:
fine